Assets/scripts/MelodyUnit.cs

```diff
@@ -103,4 +103,22 @@ public override void UnMark()
     {
         GetComponent<Renderer>().material.color = LeadingColor;
     }
+
+	public void DealDamage(MelodyUnit other, float damagePercent)
```

Comment:
@GregoryElliott This could be another place to look for why damage isn't working.